### PR TITLE
Add new realm to map

### LINKED ISSUE #637: Add new Realm (tentative)
There is a high chance we will be adding a new realm, to prepare for this i've laid out what the next realm position would be.
Colours/Name and realm details are still unknown as we don't know yet what the realm is actually going to be.

In my system the realm is nr 16 (but idk how they are numbered in code)

image

Colour: #F7DC6F

### PR DESCRIPTION
**Pull request by xpaczka**
Resolves #637 

What has been done:
- Added new realm to the island
- Added support for multi part path cutouts
- Added `useMultiRef` hook
- Removed not needed clouds

Make sure to add to `.env` to see the new realm (think it is quite universal for future additions):
- `SHOW_ADDITIONAL_REALMS="true"`

### CODE REVIEW

File: src/ui/Island/Realm.tsx [hunk 1]
```diff
@@ -47,46 +51,28 @@ export default function Realm({
...
-    islandContext.current.onRealmClick(id)
-  }
 
-  useLayoutEffect(() => {
-    const pathRealm = pathRef.current
-    const group = groupRef.current
-    const stage = pathRef.current?.getStage()
-    if (!stage || !pathRealm || !group) return () => {}
-    const defaultZ = group.zIndex()
+    if (id === "16") return // TODO: remove when realm fully implemented
```
Comment (jagodarybacka) [inline src/ui/Island/Realm.tsx:64]: I think we don't need to do it, we can open the realm modal here

File: src/ui/Island/Realm.tsx [hunk 2]
```diff
@@ -47,46 +51,28 @@ export default function Realm({
...
   const groupRef = useRef<Konva.Group>(null)
-  const pathRef = useRef<Konva.Path>(null)
   const textRef = useRef<Konva.Text>(null)
-  const imageLayerRef = useRef<Konva.Image>(null)
-  const overlayRef = useRef<Konva.Path>(null)
   const partnerLogoRef = useRef<Konva.Image>(null)
 
+  const [pathRefs, addPathRef] = useMultiRef<Konva.Path>()
+  const [, addOverlayRefs] = useMultiRef<Konva.Path>()
+  const [, addImageLayerRef] = useMultiRef<Konva.Image>()
```
Comment (jagodarybacka) [inline src/ui/Island/Realm.tsx:59]: actually why do we need to set these refs? I can't find a place where they are used after they are set. Is it internal Konva's thing?

**Review by jagodarybacka — COMMENTED** [review]
Looks great 🔥 

I think we can:
- [ ] allow to open the realm modal, take care of the UI there
- [ ] check the pin placement both on the map and on the realm modal header

File: src/ui/Island/Realm.tsx (hunk 2 above)
Comment (xpaczka) [reply src/ui/Island/Realm.tsx:59]: We don't, good catch

File: src/shared/utils/island.ts [hunk 3]
```diff
@@ -218,3 +223,34 @@ export function calculatePartnerLogoTranslate(text: string): number {
...
+      return { x: 200, y: 250 }
+    default:
+      return { x: 250, y: 250 }
+  }
+}
+
+export function getMapRealmsData() {
+  let realmsData = REALMS_MAP_DATA
+
+  if (!process.env.SHOW_ADDITIONAL_REALM && process.env.ADDITONAL_REALM_ID) {
```
[the remark this answers is not in the document]
Comment (jagodarybacka) [reply src/shared/utils/island.ts:249]: Or we can be prepared for multiple additional realms as well and just have a list in code of these while using `SHOW_ADDITIONAL_REALMS=true/false`

**Comment by jagodarybacka** [comment]
I just noticed - on the realm modal only the "main land" of the realm is visible and the mini island are missing

image

File: src/shared/constants/realms.ts [hunk 4]
```diff
@@ -54,14 +61,20 @@ export const FIGMA_FACTOR = {
...
   realm4,
   realm7,
   realm9,
   realm19,
   realm22,
 ]
 
+export const ADDITIONAL_REALMS_MAP_DATA: RealmMapData[] = [realm16]
+
+export const REALMS_MAP_DATA = process.env.SHOW_ADDITIONAL_REALMS
```
Comment (jagodarybacka) [inline src/shared/constants/realms.ts:74]: it must be `process.env.SHOW_ADDITIONAL_REALMS === "true"` becuase env vars are strings so `SHOW_ADDITIONAL_REALMS="false"`will break it

**Review by jagodarybacka — CHANGES_REQUESTED** [review]
One small thing and we are good to go. If we will decide to make other part of the map active I think it will be easier to do it in the next PR.

**Review by jagodarybacka — APPROVED** [review]
:shipit:

**Comment by jagodarybacka** [comment]
Will merge it tomorrow to not bloat the batch for QA before the XP drop

File: src/shared/components/RealmCutout/RealmCutout.tsx [hunk 5]
```diff
@@ -44,41 +46,49 @@ export default function RealmCutout() {
...
+                  <use
+                    href={`#${pathCutoutXref}_${index}`}
+                    fill="#fff"
+                    transform={`translate(${pathData.x}, ${pathData.y})`}
+                  />
+                </mask>
+              </Fragment>
+            ))}
+          </defs>
+          {pathData.paths.map((_, index) => (
```
Comment (ioay) [inline src/shared/components/RealmCutout/RealmCutout.tsx:67]: Let's put it in one iteration of array mapping.

**Review by jagodarybacka — APPROVED** [review]

**Pull request merged**